### PR TITLE
fix/standardise storybook control option and default options

### PR DESCRIPTION
**Pull request by mJarsater**
**Describe pull-request**  
This PR standardises the controls and default options for all components.

**Solving issue**  
Fixes: [AB#2483](https://dev.azure.com/scaniadigitaldesignsystem/3d9eb9c6-0045-473d-bc1d-e842dd779200/_workitems/edit/2483)

**How to test**  
1. Go to storybook link below
2. Check in all components and try out the controls.

### NOTE
This PR also replaces svgs with out own icons, and gives the user the option of having either a webcomponent or native one.

### CODE REVIEW

File: tegel/src/components/stepper/stepper.stories.js [hunk 1]
```diff
@@ -11,139 +11,121 @@ export default {
       description: 'Size of the stepper',
       control: {
         type: 'radio',
-        options: ['Default', 'Small'],
       },
+      options: ['Default', 'Small'],
       defaultValue: 'Default',
```
Comment (alexandereneroth) [inline tegel/src/components/stepper/stepper.stories.js:16]: 🕵️

File: tegel/src/components/stepper/stepper.stories.js [hunk 2]
```diff
@@ -11,139 +11,121 @@ export default {
...
     },
     style: {
       name: 'Style',
       description: 'Style of the stepper',
       control: {
         type: 'radio',
-        options: ['Default', 'Text on side', 'Vertical'],
       },
+      options: ['Default', 'Text on side', 'Vertical'],
       defaultValue: 'Default',
```
Comment (alexandereneroth) [inline tegel/src/components/stepper/stepper.stories.js:25]: 🕵️

File: tegel/src/components/stepper/stepper.stories.js [hunk 3]
```diff
@@ -11,139 +11,121 @@ export default {
...
-    label: {
-      name: 'Label',
-      description: 'Show or hide the label',
+    showLabel: {
+      name: 'Show label text',
+      description: 'Show or hide the label text',
+      control: {
+        type: 'boolean',
+      },
+      defaultValue: true,
```
Comment (alexandereneroth) [inline tegel/src/components/stepper/stepper.stories.js:33]: 🕵️

File: tegel/src/components/textarea/textarea.stories.js [hunk 4]
```diff
@@ -54,7 +54,6 @@ export default {
         options: ['No label', 'Inside', 'Outside'],
       },
       defaultValue: 'No label',
```
Comment (alexandereneroth) [inline tegel/src/components/textarea/textarea.stories.js:56]: defaultValue 👆

File: tegel/src/components/toast/toast.stories.ts [hunk 5]
```diff
@@ -12,68 +12,100 @@ export default {
...
+  return formatHtmlPreview(
     `
-  <div class="sdds-toast sdds-toast-${args.toastType}">
+    <style>
+      ${
+        iconType === 'Native'
+          ? `@import url('https://cdn.digitaldesign.scania.com/icons/webfont/css/sdds-icons.css');
+      i {
+        font-size: 20px;
+        color: var(--sdds-${colorLooup[toastType]})
```
Comment (alexandereneroth) [inline tegel/src/components/toast/toast.stories.ts:70]: Missing semicolon ;

File: tegel/src/components/toast/toast.stories.ts [hunk 6]
```diff
@@ -12,68 +12,100 @@ export default {
...
+      }
+      `
+          : `
+      sdds-icon{
+        color: var(--sdds-${colorLooup[toastType]})
```
Comment (alexandereneroth) [inline tegel/src/components/toast/toast.stories.ts:75]: Missing semicolon ;

**Review by alexandereneroth — CHANGES_REQUESTED** [review]
Awesome work! Just spotted a few remaining `defaultValue`, if you fix those i'll approve it

**Comment by mJarsater** [comment]
> Awesome work! Just spotted a few remaining `defaultValue`, if you fix those i'll approve it

Thanks! Yeah im working on those in a separate branch that I'm gonna merge into this one

File: tegel/src/components/button/button-webcomponent.stories.ts [hunk 7]
```diff
@@ -87,55 +87,96 @@ export default {
...
+    fullbleed: false,
+    disabled: false,
+    onlyIcon: false,
+    icon: 'none',
   },
 };
 
-const WebComponentTemplate = ({ onlyIcon, size, variant, btnType, fullbleed, disabled, icon, text = 'Button' }) => {
+const WebComponentTemplate = ({ onlyIcon, size, variant, btnType, fullbleed, disabled, icon, iconType, text = 'Button' }) => {
+  const btnTypeLookUp = {
```
[the remark this answers is not in the document]
Comment (mJarsater) [reply tegel/src/components/button/button-webcomponent.stories.ts:114]: Yeah I guess this goes for all lookup objects. I can do them all in one commit.

**Review by alexandereneroth — CHANGES_REQUESTED** [review]
Some final few things 👍

File: tegel/src/components/stepper/stepper.stories.js [hunk 8]
```diff
@@ -11,139 +11,123 @@ export default {
...
-
-  switch (label) {
-    case 'Hide':
-      showLabel = false;
-      break;
-    default:
-      break;
-  }
+const Template = ({ size, style, showLabel, iconType }) => {
+  let sizeClass = size === 'Small' ? 'sdds-stepper--small' : '';
```
[the remark this answers is not in the document]
Comment (mJarsater) [reply tegel/src/components/stepper/stepper.stories.js:50]: Nice!

File: tegel/src/components/block/block.stories.ts [hunk 9]
```diff
@@ -1,44 +1,41 @@
...
+    Variant: 'variant',
+  };
+
+  return formatHtmlPreview(
+    `
     <div class="sdds-demo-block"> 
-      <div class="${args.type === 'variant' ? 'sdds-block sdds-block__variant' : 'sdds-block'}">
+      <div class="${typeLookup[type] === 'variant' ? 'sdds-block sdds-block__variant' : 'sdds-block'}">
         <p>Content be here...</p>
-        <div class="sdds-block">
```
[the remark this answers is not in the document]
Comment (mJarsater) [reply tegel/src/components/block/block.stories.ts:33]: Not sure I get what you mean here?

[the remark this answers is not in the document]
Comment (mJarsater) [reply tegel/src/components/block/block.stories.ts:33]: Never mind! Pushed a change.

**Review by alexandereneroth — APPROVED** [review]
Great work once again!

File: tegel/src/components/toast/toast.scss [hunk 10]
```diff
@@ -10,8 +10,8 @@
   width: 88rem;
   min-height: var(--sdds-spacing-element-48);
 
-  &-icon svg {
-    margin: 4rem 0 0 3rem;
+  &-icon {
+    margin: 13.5px 0 0 3rem;
```
Comment (theJohnnyMe) [inline tegel/src/components/toast/toast.scss:14]: Please try to replace this one with:
```suggestion
    margin: 14px 0 0 12px;
```

I am not sure if 0.5px works on screens + let's have px values everywhere for consistency.

File: tegel/src/components/toast/toast.stories.ts [hunk 11]
```diff
@@ -9,71 +9,102 @@ export default {
...
       description: 'Adds a CTA link',
     },
+    iconType: {
+      name: 'Icon type',
+      description: 'Native/Webcomponent',
+      control: {
+        type: 'radio',
+      },
+      options: ['Native', 'Webcomponent'],
+    },
```
Comment (theJohnnyMe) [inline tegel/src/components/toast/toast.stories.ts:32]: Please set one of them as default as the story ends up with non of them pre-selected. I would say when in doubt, make the component version default.

**Review by theJohnnyMe — CHANGES_REQUESTED** [review]

**Comment by theJohnnyMe** [comment]
I found really tiny things, I tried to go commit by commit. 
I found some more stuff but they are not related to your changes. all in all, awesome work on standardizing buddy! 🚀

**Comment by mJarsater** [comment]
> I found really tiny things, I tried to go commit by commit. I found some more stuff but they are not related to your changes. all in all, awesome work on standardizing buddy! 🚀

Thanks! Great, I re-requested your feedback now, after I fixed your suggested changes 😄

**Review by theJohnnyMe — APPROVED** [review]
Gods work!

**Pull request merged**